File: templates/entity/entity.ts
import { ApiProperty } from '@nestjs/swagger';
import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';

@Entity()
export class {{name}} {
    {{#each columns}}
    {{#if isPrimary}}
    @PrimaryGeneratedColumn()
    {{/if}}
    {{#unless isPrimary}}
    @Column()
    {{/unless}}
    @ApiProperty()
    {{name}} : {{type}}

    {{/each}}
}

File: cli/lib/generators/spring-boot-jpa/template/pojo/entity.java
package {{serverPackageName}}.pojo;

import lombok.Data;

import javax.persistence.*;

{{#if (hasDate columns)}}
import java.time.LocalDateTime;
{{/if}}

@Entity
@Data
public class {{name}} {
    {{#each columns}}
    {{#if isPrimary}}
    @Id
    @GeneratedValue
    {{/if}}
    {{#unless isPrimary}}
    
    {{#if hasForeignKey}}
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "{{joinColumn}}", nullable = false)
    {{/if}}
    {{#unless hasForeignKey}}
    @Column
    {{/unless}}
    {{/unless}}
    private {{javaType type}} {{name}};
    {{/each}}
}
